src/Compilers/CSharp/Test/Semantic/Semantics/NativeIntegerTests.cs

```diff
@@ -6251,5 +6251,183 @@ .maxstack  2
   IL_0003:  ret
 }");
         }
+
+        [Fact]
+        public void ConstantFolding()
+        {
+            constantFolding("nint", "const nint A = -2147483648;", "+A", "-2147483648");
+            constantFolding("nint", "const nint A = 2147483647;", "+A", "2147483647");
+            constantFolding("nuint", "const nuint A = 0;", "+A", "0");
+            constantFolding("nuint", "const nuint A = 4294967295;", "+A", "4294967295");
+
+            constantFolding("nint", "const nint A = -2147483648;", "-A", null);
+            constantFolding("nint", "const nint A = -2147483647;", "-A", "2147483647");
+            constantFolding("nint", "const nint A = 2147483647;", "-A", "-2147483647");
+            constantFolding("nuint", "const nuint A = 0;", "-A", null, Diagnostic(ErrorCode.ERR_AmbigUnaryOp, "-A").WithArguments("-", "nuint")); // PROTOTYPE: Should report ERR_NoImplicitConvCast
+            constantFolding("nuint", "const nuint A = 1;", "-A", null, Diagnostic(ErrorCode.ERR_AmbigUnaryOp, "-A").WithArguments("-", "nuint")); // PROTOTYPE: Should report ERR_NoImplicitConvCast
+            constantFolding("nuint", "const nuint A = 4294967295;", "-A", null, Diagnostic(ErrorCode.ERR_AmbigUnaryOp, "-A").WithArguments("-", "nuint")); // PROTOTYPE: Should report ERR_NoImplicitConvCast
+
+            constantFolding("nint", "const nint A = -2147483648;", "~A", "2147483647");
+            constantFolding("nint", "const nint A = 2147483647;", "~A", "-2147483648");
+            constantFolding("nuint", "const nuint A = 0;", "~A", "4294967295"); // PROTOTYPE: Should this be an error?
```

Comment:
I don't think these are correct.  These assume a `nint` is 32 bits, which is only correct on some platforms.  Since we don't know how many bits there are in a `nint`, we cannot fold the `~` operation.